sdk/src/client/secret/mnemonic.rs

```diff
@@ -98,6 +100,66 @@ impl From<Mnemonic> for MnemonicSecretManager {
     }
 }
 
+/// A mnemonic (space separated list of words) that allows to create a seed from.
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Zeroize, ZeroizeOnDrop)]
+pub struct Mnemonic(String);
+
+impl Mnemonic {
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+}
+
+impl TryFrom<String> for Mnemonic {
+    type Error = Error;
+
+    fn try_from(mut value: String) -> Result<Self, Self::Error> {
+        // trim because empty spaces could create a different seed https://github.com/iotaledger/crypto.rs/issues/125
+        let trimmed = value.trim();
+        // first we check if the mnemonic is valid to give meaningful errors
+        if let Err(err) = crypto::keys::bip39::wordlist::verify(trimmed, &crypto::keys::bip39::wordlist::ENGLISH) {
+            value.zeroize();
+            Err(crate::client::Error::InvalidMnemonic(format!("{err:?}")))
+        } else {
+            let mnemonic = trimmed.to_string();
+            value.zeroize();
+            Ok(Self(mnemonic))
+        }
+    }
+}
+
+pub trait MnemonicLike: Send {
+    fn to_mnemonic(self) -> Result<Mnemonic, Error>;
+    fn as_str(&self) -> &str;
+}
+
+impl MnemonicLike for Mnemonic {
+    fn to_mnemonic(self) -> Result<Mnemonic, Error> {
+        Ok(self)
+    }
+
+    fn as_str(&self) -> &str {
+        self.as_str()
+    }
+}
+
+impl MnemonicLike for String {
+    fn to_mnemonic(self) -> Result<Mnemonic, Error> {
+        Mnemonic::try_from(self)
+    }
+
+    fn as_str(&self) -> &str {
+        self.as_str()
+    }
+}
```

Comment:
While we're at it, you could impl it for lists?

Reply:
yeah, do you think it's useful to have?

Reply:
Could be. I feel like it makes sense mostly.

Reply:
Done